marklogic-data-hub/src/main/java/com/marklogic/hub/ArtifactManager.java

```diff
@@ -0,0 +1,102 @@
+     */
+    public JsonNode getArtifacts(String artifactType);
+
+    /**
+     * Updates an artifact of a given type and name.
+     *
+     * @param artifactType - type of artifact the operation is dealing with
+     * @param artifactName - identifier of the artifact
+     * @param artifactJson - JSON of artifact to persist
+     * @return JsonNode of the artifact matching a type
+     */
+    public JsonNode updateArtifact(String artifactType, String artifactName, JsonNode artifactJson) throws IOException;
+
+    /**
+     * Retrieves an artifact of a given type and name.
+     *
+     * @param artifactType - type of artifact the operation is dealing with
+     * @param artifactName - identifier of the artifact
+     * @return JsonNode of the artifact matching a type and name
+     */
+    public JsonNode getArtifact(String artifactType, String artifactName);
+
+    /**
+     * Deletes an artifact of a given type and name.
+     *
+     * @param artifactType - type of artifact the operation is dealing with
+     * @param artifactName - identifier of the artifact
+     * @return JsonNode
+     */
+    public JsonNode deleteArtifact(String artifactType, String artifactName) throws IOException;
+
+    /**
+     * Validates an artifact of a given type and name.
+     *
+     * @param artifactType - type of artifact the operation is dealing with
+     * @param artifactName - identifier of the artifact
+     * @param artifactJson - JSON of artifact to persist
+     * @return JsonNode with validation details
+     */
+    public JsonNode validateArtifact(String artifactType, String artifactName, JsonNode artifactJson);
+
+    /**
+     * Retrieves settings of an artifact of a given type and name.
+     *
+     * @param artifactType - type of artifact the operation is dealing with
+     * @param artifactName - identifier of the artifact
+     * @return JsonNode of the artifact settings matching a type and name
+     */
+    public JsonNode getArtifactSettings(String artifactType, String artifactName);
+
+    /**
+     * Updates settings of an artifact of a given type and name.
+     *
+     * @param artifactType - type of artifact the operation is dealing with
+     * @param artifactName - identifier of the artifact
+     * @param settings - JSON of artifact settings to persist
+     * @return JsonNode of the artifact settings matching a type and name
+     */
+    public JsonNode updateArtifactSettings(String artifactType, String artifactName, JsonNode settings);
+
+    /**
+     * Provides path to artifact in project for an artifact of a given type, name, and (optionally) version.
+     *
+     * @param artifactType - type of artifact the operation is dealing with
+     * @param artifactName - identifier of the artifact
+     * @param artifactVersion - version of artifact (optional)
+     * @return Path to artifact in project
+     */
+    public Path buildArtifactProjectLocation(String artifactType, String artifactName, String artifactVersion);
```

Comment:
It looks like the only client of this is an automated test. For that reason, I don't think it should be the interface yet. The fewer methods in an interface, the better. The method in the impl can be public if needed, and the test can just call it directly instead of depending on the interface.